File: backend/src/main/java/es/dimecresalessis/scoutbase/user/infrastructure/web/dto/UserDto.java
package es.dimecresalessis.scoutbase.user.infrastructure.web.dto;

import jakarta.validation.constraints.NotBlank;
import lombok.AllArgsConstructor;
import lombok.Data;

import java.util.UUID;

@Data
@AllArgsConstructor
public class UserDto {
    private UUID id;
    @NotBlank
    private String username;
    @NotBlank
    private String password;
    private String role;
}

File: mobileFrontend/app/src/main/java/com/empresa/scoutbase/data/remote/dto/UserDto.kt
package com.empresa.scoutbase.data.remote.dto

data class UserByUsernameResponse(
    val success: Boolean,
    val message: String,
    val data: UserData?,
    val sessionId: String?,
    val timestamp: String?
)

data class UserData(
    val id: String,
    val username: String,
    val password: String,
    val role: String
)
